src/main/java/duke/utils/Parser.java

```diff
@@ -0,0 +1,90 @@
+package duke.utils;
+
+import duke.task.Deadlines;
+import duke.task.Events;
+import duke.task.Task;
+import duke.task.Todos;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Parser {
+
+    public static Date stringToDate(String text) {
+        Date date = null;
+        SimpleDateFormat formatter = new SimpleDateFormat(Constants.DATE_FORMAT_1);
+        try {
+            date = formatter.parse(text);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        return date;
+    }
+
+    public static boolean checkDateFormat(String text) {
```

Comment:
I believe you can import a library/package for date and time, which comes with their own exceptions. This will make your life much easier. Perhaps try this link: https://www.journaldev.com/16896/java-localdate